includes/class-wc-stripe-webhook-handler.php

```
@@ -1527,43 +1601,70 @@ public function process_checkout_session_success( object $notification ): void {
+		// The session ID may not be stored yet if the webhook arrived before
+		// process_payment_with_checkout_session completed (race condition).
 		$order = WC_Stripe_Helper::get_order_by_checkout_session_id( $checkout_session->id );
-		if ( ! $order instanceof \WC_Order ) {
-			try {
-				$this->handle_agentic_checkout_session( $notification );
-			} finally {
-				WC_Stripe_Database_Cache::delete( $lock_key );
+
+		// Release the lock — it served its purpose of de-duplicating concurrent deliveries of this
+		// webhook. Deferred processing is idempotent (status checks prevent double-processing).
+		WC_Stripe_Database_Cache::delete( $lock_key );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Database_Cache' in method 'process_checkout_session_success'. (undefined)

(StaticAccess)